dreadnode/cli/docker.py

```diff
@@ -0,0 +1,387 @@
+import re
+import subprocess  # nosec
+import typing as t
+
+from pydantic import AliasChoices, BaseModel, Field
+
+from dreadnode.common_types import UNSET, Unset
+from dreadnode.constants import (
+    DEFAULT_DOCKER_REGISTRY_IMAGE_TAG,
+    DEFAULT_DOCKER_REGISTRY_LOCAL_PORT,
+    DEFAULT_DOCKER_REGISTRY_SUBDOMAIN,
+    DEFAULT_PLATFORM_BASE_DOMAIN,
+)
+from dreadnode.logging_ import print_info
+from dreadnode.user_config import ServerConfig
+
+DockerContainerState = t.Literal[
+    "running", "exited", "paused", "restarting", "removing", "created", "dead"
+]
+
+
+class DockerError(Exception):
+    pass
+
+
+class DockerImage(str):  # noqa: SLOT000
```

Comment:
Moved this to follow our GithubRepo pattern with a bunch of tests. Image parsing will be a pretty large mech in the broader capability stack.